WordPress/Classes/ViewRelated/Reader/Tab Navigation/WPTabBarController+ReaderTabNavigation.swift

```diff
@@ -27,17 +27,31 @@ extension WPTabBarController {
     }
 
     @objc func makeReaderTabViewController() -> ReaderTabViewController {
-        return ReaderTabViewController(viewModel: readerTabViewModel) { [unowned self] viewModel in
+        return ReaderTabViewController(viewModel: readerTabViewModel) { [weak self] viewModel in
+            guard let self else {
+                return ReaderTabView(viewModel: viewModel)
+            }
             return self.makeReaderTabView(viewModel)
         }
     }
 
     @objc func makeReaderTabViewModel() -> ReaderTabViewModel {
         let viewModel = ReaderTabViewModel(
-            readerContentFactory: { [unowned self] in
-                self.makeReaderContentViewController(with: $0)
+            readerContentFactory: { [weak self] content in
+                if content.topicType == .discover, let topic = content.topic {
+                    let controller = ReaderCardsStreamViewController.controller(topic: topic)
+                    controller.shouldShowCommentSpotlight = self?.readerTabViewModel.shouldShowCommentSpotlight ?? false
+                    return controller
+                } else if let topic = content.topic {
+                    return ReaderStreamViewController.controllerWithTopic(topic)
+                } else {
+                    return ReaderStreamViewController.controllerForContentType(content.type)
+                }
```

Comment:
I couldn't think of a great way to fix this without having a chain reaction. If we made the return here optional and returned `nil`, then we'd need to update a few different places. I figured for now, we can get away with copying the body of the function call, `makeReaderContentViewController` into this closure.